src/metabase/lib/common.cljc

```diff
@@ -0,0 +1,48 @@
+(ns metabase.lib.common
+  (:require
+   [metabase.lib.dispatch :as lib.dispatch]
+   [metabase.lib.field :as lib.field]
+   #_{:clj-kondo/ignore [:unused-namespace]}
+   [metabase.lib.options :as lib.options])
+  #?(:cljs (:require-macros [metabase.lib.common])))
+
+(defmulti ->op-arg
+  "Ensures that clause arguments are properly unwrapped"
+  {:arglists '([query stage-number x])}
+  (fn [_query _stage-number x]
+    (lib.dispatch/dispatch-value x)))
+
+(defmethod ->op-arg :default
+  [_query _stage-number x]
+  x)
```

Comment:
If we allow aggregating expressions, we have to recurse into clauses. (I do this for all vectors using `mapv`.)